src/test/java/com/gravity9/mongocdc/ChangeStreamTest.java

```diff
@@ -45,14 +55,15 @@ void givenProperConfiguration_shouldStartManagerAndRegisterListener() throws Exc
 		// Wait for manager to start
 		Thread.sleep(1000);
 
-		var collection = MongoClientProvider.createClient(CONN_URI).getDatabase(DB_NAME).getCollection(COLL_NAME);
+		var collection = MongoClientProvider.createClient(getConnectionUri()).getDatabase(getDatabaseName()).getCollection(getTestCollectionName());
+
 		Document testDoc = new Document("testValue", 123);
-		collection.insertOne(testDoc);
+		InsertOneResult insertOneResult = collection.insertOne(testDoc);
+		log.info("Inserted document: {}", insertOneResult.getInsertedId());
 
-		// Wait for CDC event to arrive
-		Thread.sleep(1000);
+		List<ChangeStreamDocument<Document>> events;
+		events = waitForEvents(listener);
```

Comment:
change it to `List<ChangeStreamDocument<Document>> events = waitForEvents(listener);`

Reply:
Fixed